engine/buff/buff.cpp

```diff
@@ -627,6 +628,7 @@ buff_t::buff_t( sim_t* sim, player_t* target, player_t* source, util::string_vie
     buff_duration_multiplier( 1.0 ),
     default_chance( 1.0 ),
     manual_chance( -1.0 ),
+    time_modifier( 0.0 ),
```

Comment:
Tracking the modifier rather than multiplier is very unusual as far as simc goes. Presumably, the time mod stuff stacks multiplicatively (though that might be hard to test right now). But even if Blizz never stacks these, it might still be useful in simc.

Reply:
I can do stacking, but, if how [echos of elisande](https://www.wowhead.com/spell=320919/echoes-of-elisande) stacked is any indication, it'll be additive and not multiplicative (though that was multiple stacks of an aura effecting an effect value, rather than multiple effect values combining). But I'm not sure we can know which until we have a live example. Though I'm thinking it's probably be additive, since I'm *pretty sure* time warp (with a base value of 0), didn't break people out of jailer stun (value of -99), but I can't say that for sure.

Reply:
Echoes of Elisande doesn't really tell us anything. 2 stacks of a +10% damage buff give you +20% damage, even though damage buffs stack multiplicatively. You'd need to find two different effects to combine (not one effect with multiple stacks) to tell them apart. I'm not sure how being additive/multiplicative has anything to do with Time Warp breaking people out of Jailer stun. If it's additive then you have `0 + (-99) = -99` mod, if it's multiplicative you have `(1 + 0) * (1 + (-99)/100) = 0.01` multiplier; both of which don't indicate you should be broken out. I suppose you could go to Gul'dan, use the extra action button and check the time mod by using one of the buff API functions.

Reply:
So the magistrix's [time stop](https://www.wowhead.com/spell=208944/time-stop) allowed me to witness a modRate of 10000 applied to my paused buffs (two 100x slows multiplied together). While [time dilation](https://www.wowhead.com/spell=210339/time-dilation) saw my paused buff rates become 76.923080444336, which is equivalent to a 100x slowdown and a 30% speedup being multiplied together, so multiplicative stacking it is. Good to know that while it seems like individual effects seem to be limited to 100x slower, though, the combined modifier can still exceed that.